src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/SyndicationFeedFormatter.cs

```diff
@@ -2,191 +2,169 @@
-            {
-                return _feed;
-            }
-        }
+        public SyndicationFeed Feed { get; private set; }
 
         public TryParseUriCallback UriParser { get; set; } = DefaultUriParser;
 
         // Different DateTimeParsers are needed for Atom and Rss so can't set inline
         public TryParseDateTimeCallback DateTimeParser { get; set; }
 
-        internal virtual TryParseDateTimeCallback GetDefaultDateTimeParser()
-        {
-            return NotImplementedDateTimeParser;
-        }
+        internal virtual TryParseDateTimeCallback GetDefaultDateTimeParser() => NotImplementedDateTimeParser;
 
         private bool NotImplementedDateTimeParser(XmlDateTimeData XmlDateTimeData, out DateTimeOffset dateTimeOffset)
         {
-            dateTimeOffset = default(DateTimeOffset);
+            dateTimeOffset = default;
             return false;
         }
 
-        public abstract string Version
-        { get; }
+        public abstract string Version { get; }
 
         public abstract bool CanRead(XmlReader reader);
 
         public abstract void ReadFrom(XmlReader reader);
 
         public override string ToString()
         {
-            return String.Format(CultureInfo.CurrentCulture, "{0}, SyndicationVersion={1}", this.GetType(), this.Version);
+            return string.Format(CultureInfo.CurrentCulture, "{0}, SyndicationVersion={1}", GetType(), Version);
         }
 
         public abstract void WriteTo(XmlWriter writer);
 
         internal static protected SyndicationCategory CreateCategory(SyndicationFeed feed)
         {
-            if (feed == null)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("feed");
-            }
-            return GetNonNullValue<SyndicationCategory>(feed.CreateCategory(), SR.FeedCreatedNullCategory);
+            Debug.Assert(feed != null);
+            return GetNonNullValue(feed.CreateCategory(), SR.FeedCreatedNullCategory);
         }
 
         internal static protected SyndicationCategory CreateCategory(SyndicationItem item)
         {
             if (item == null)
             {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("item");
+                throw new ArgumentNullException(nameof(item));
             }
-            return GetNonNullValue<SyndicationCategory>(item.CreateCategory(), SR.ItemCreatedNullCategory);
+
+            return GetNonNullValue(item.CreateCategory(), SR.ItemCreatedNullCategory);
         }
 
         internal static protected SyndicationItem CreateItem(SyndicationFeed feed)
         {
-            if (feed == null)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("feed");
-            }
-            return GetNonNullValue<SyndicationItem>(feed.CreateItem(), SR.FeedCreatedNullItem);
+            Debug.Assert(feed != null);
+            return GetNonNullValue(feed.CreateItem(), SR.FeedCreatedNullItem);
```

Comment:
It's possible to be passed a null here too. Revert the null check exception change.